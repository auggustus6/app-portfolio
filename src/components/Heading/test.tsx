/* eslint-disable no-undef */

import React from 'react';
import Title from '.';
import { render } from '../../utils/tests/helpers';

it('should render correctly', async () => {
  const { getByText, toJSON } = render(<Title label="Titulo" />);
  const text = getByText(/Titulo/i);
  expect(text).toBeDefined();
  expect(toJSON()).toMatchSnapshot();
});

it('should render text color black by default', async () => {
  const { getByText } = render(<Title label="Titulo" />);
  const text = getByText(/Titulo/i);
  expect(text).toHaveStyleRule('color', '#333');
});

it('should render font size large by default', async () => {
  const { getByText } = render(<Title label="Titulo" />);
  const text = getByText(/Titulo/i);

  expect(text).toHaveStyleRule('font-size', 22);
});

it('should render font size medium', async () => {
  const { getByText } = render(<Title label="Titulo" size="medium" />);
  const text = getByText(/Titulo/i);

  expect(text).toHaveStyleRule('font-size', 18);
});

it('should render font size small', async () => {
  const { getByText } = render(<Title label="Titulo" size="small" />);
  const text = getByText(/Titulo/i);

  expect(text).toHaveStyleRule('font-size', 16);
});

it('should align the font in the left ', async () => {
  const { getByText } = render(<Title label="Titulo" size="small" />);
  const text = getByText(/Titulo/i);

  expect(text).toHaveStyleRule('text-align', 'left');
});
