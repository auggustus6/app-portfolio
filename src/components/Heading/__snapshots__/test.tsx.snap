// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Text
  align="left"
  colorLabel="#333"
  size="large"
  strong={false}
  style={
    Array [
      Object {
        "color": "#333",
        "fontFamily": "Poppins-Regular",
        "fontSize": 24,
        "marginTop": 18,
        "textAlign": "left",
      },
    ]
  }
>
  Titulo
</Text>
`;
